src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -10744,12 +10919,12 @@ instruct MoveL2D_reg_reg(regD dst, rRegL src) %{
 
 // =======================================================================
 // fast clearing of an array
-instruct rep_stos(rcx_RegL cnt, rdi_RegP base, regD tmp, rax_RegI zero,
+instruct rep_stos(rcx_RegL cnt, rdi_RegP base, regD tmp, kReg ktmp, rax_RegI zero,
```

Comment:
I'm curious how it works w/o AVX512 considering`rep_stos` and others are not AVX512-specific AD instructions.

Reply:
Instructions using kReg temp are all guarded by UseAVX > 2 checks. Thus they are never JITed.

Reply:
But then RA has to allocate a k-register irrespective of whether AVX512 is supported or not, doesn't it?

Reply:
Yes, allocation will still happen but it will not seep into any JITed instruction. Reg_Defs/Operands/RegMask/RegClass are not guarded by target checks.

Reply:
Ok, now I understand how it is intended to work. Thanks for the clarifications.

I'm not fully satisfied with how it works. I'm in favor of avoiding using any functionality which is not supported by hardware. (And that's currently the case.) But considering there's no simple way to fix it right now, I'm fine with addressing it separately as a follow-up fix. 

For now, please, put extensive comments in relevant places describing why `kReg` is used unconditionally. As the code shaped now, it's confusing at the very least.

In the longer term, IMO the proper way to implement that is for ADLC to support conditional operands which depend on runtime conditions and are matched to `*noreg`s when the condition is false.